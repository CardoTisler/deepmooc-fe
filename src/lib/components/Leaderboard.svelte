<script lang="ts">
	import type { LeaderboardRow } from '$lib/types';
	export let rows: LeaderboardRow[];
	let style = '';
	export { style as class };
</script>

<div
	class="rounded-xl bg-gray-100 border-2 border-gray-200 h-36 flex flex-col justify-evenly container max-w-xl {style}"
>
	{#each rows as { index, name, score, bold }, i}
		<div class="grid grid-cols-3 gap-4 w-full">
			<div class="flex justify-start pl-4"><p class={bold ? 'font-bold' : ''}>{index}.</p></div>
			<div class="flex justify-center"><p class={bold ? 'font-bold' : ''}>{name}</p></div>
			<div class="flex justify-end pr-4"><p class={bold ? 'font-bold' : ''}>{score}</p></div>
		</div>
		{#if i == 0}
			<div class="grid grid-cols-3 gap-4 w-full">
				<div class="flex justify-start pl-4">
					<p>&#8942;</p>
				</div>
				<div class="flex justify-center">
					<p>&#8942;</p>
				</div>
				<div class="flex justify-end pr-4">
					<p>&#8942;</p>
				</div>
			</div>
		{/if}
	{/each}
</div>

<style>
	.container {
		min-width: 360px;
	}
</style>
