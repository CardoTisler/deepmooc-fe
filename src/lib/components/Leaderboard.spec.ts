import type { LeaderboardRow } from '$lib/types';
import { render } from '@testing-library/svelte';
import Leaderboard from './Leaderboard.svelte';

describe('Leaderboard.svelte', () => {
	it('should render component', () => {
		const mockRows: LeaderboardRow[] = [
			{
				index: 1,
				name: 'TEST NAME',
				score: 100,
				bold: true
			}
		];

		const { container } = render(Leaderboard, { rows: mockRows });
		expect(container).toMatchSnapshot();
	});
});
