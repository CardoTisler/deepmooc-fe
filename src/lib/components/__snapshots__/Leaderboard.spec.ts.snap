// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Leaderboard.svelte > should render component 1`] = `
<body>
  <div>
    <div
      class="rounded-xl bg-gray-100 border-2 border-gray-200 h-36 flex flex-col justify-evenly container max-w-xl dark:bg-black dark:border-yellow-300  s-2C3Pk0x1uTjp"
    >
      <div
        class="grid grid-cols-3 gap-4 w-full s-2C3Pk0x1uTjp"
      >
        <div
          class="flex justify-start pl-4 s-2C3Pk0x1uTjp"
        >
          <p
            class="font-bold s-2C3Pk0x1uTjp"
          >
            1
            .
          </p>
        </div>
         
        <div
          class="flex justify-center s-2C3Pk0x1uTjp"
        >
          <p
            class="font-bold s-2C3Pk0x1uTjp"
          >
            TEST NAME
          </p>
        </div>
         
        <div
          class="flex justify-end pr-4 s-2C3Pk0x1uTjp"
        >
          <p
            class="font-bold s-2C3Pk0x1uTjp"
          >
            100
          </p>
        </div>
      </div>
       
      <div
        class="grid grid-cols-3 gap-4 w-full s-2C3Pk0x1uTjp"
      >
        <div
          class="flex justify-start pl-4 s-2C3Pk0x1uTjp"
        >
          <p
            class="s-2C3Pk0x1uTjp"
          >
            ⋮
          </p>
        </div>
         
        <div
          class="flex justify-center s-2C3Pk0x1uTjp"
        >
          <p
            class="s-2C3Pk0x1uTjp"
          >
            ⋮
          </p>
        </div>
         
        <div
          class="flex justify-end pr-4 s-2C3Pk0x1uTjp"
        >
          <p
            class="s-2C3Pk0x1uTjp"
          >
            ⋮
          </p>
        </div>
         
      </div>
      
    </div>
    <!--&lt;Leaderboard&gt;-->
  </div>
</body>
`;
